hazelcast/src/main/java/com/hazelcast/map/AbstractEntryProcessor.java

```diff
@@ -71,4 +75,16 @@ public void processBackup(Map.Entry<K, V> entry) {
             process(entry);
         }
     }
+
+    private class HazelcastInstanceAwareEntryBackupProcessorImpl extends EntryBackupProcessorImpl
+            implements HazelcastInstanceAware {
+
+        @Override
+        public void setHazelcastInstance(HazelcastInstance hazelcastInstance) {
+            final AbstractEntryProcessor<K, V> outer = AbstractEntryProcessor.this;
+            if (outer instanceof HazelcastInstanceAware) {
```

Comment:
This check is obligatory, isn't it? Since the class that extends the `AbstractEntryProcessor` may not implement the `HazelcastInstanceAware`, so the cast may fail.